src/Pages/SignUp/SignUpInfo.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from "react";
+import StatusView from "./StatusView";
+import config from "../../config.json";
+import "./SignUpInfo.scss";
+
+class SignUpInfo extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      id: {
+        state: false,
+        serverMsg: "",
+        value: "",
+      },
+      email: {
+        state: false,
+        value: "",
+      },
+      password: {
+        state: false,
+        value: "",
+      },
+      passwordCheck: {
+        state: false,
+        value: "",
+        check: (value) => {
+          return this.state.password.value === value ? true : false;
+        },
+      },
+      userName: {
+        state: false,
+        value: "",
+      },
+    };
+  }
+
+  validationfilter = (type, value) => {
+    const filter = {
+      id: () => (/^[A-Za-z]{1}[A-Za-z0-9]{5,19}$/.test(value) ? true : false),
+      email: () => (/^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i.test(value) ? true : false),
+      password: () => (/^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,19}$/.test(value) ? true : false),
+      passwordCheck: () => (this.state.password.value === value ? true : false),
+      userName: () => (/^[가-힣]{2,4}|[a-zA-Z]{2,10}\s[a-zA-Z]{2,10}$/.test(value) ? true : false),
+    };
+    return filter[type]();
```

Comment:
함수형태로 하신 이유가 있을까요?

Reply:
함수로 형태로  하지 않으면 validation 체크를 할 때 if else else 이렇게 많이 늘어 나게 됩니다
함수로 만들어 놓으면 쉽게 접근이 가능하고 여러 곳에서 재사용이 가능하여서 이렇게 만들어 놓았습니다 
물론 지금은 회원가입의 한정적이지만요~

Reply:
```suggestion
 validationfilter = (type, value) => {
    const filter = {
      id: /^[A-Za-z]{1}[A-Za-z0-9]{5,19}$/.test(value) ? true : false,
      email: /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i.test(value) ? true : false,
      password: /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,19}$/.test(value) ? true : false,
      passwordCheck: this.state.password.value === value ? true : false,
      userName: /^[가-힣]{2,4}|[a-zA-Z]{2,10}\s[a-zA-Z]{2,10}$/.test(value) ? true : false,
    };
    return filter[type];
 }
```

그런 이유시라면 이 형태로 하시는 게 나을 것 같네요! 굳이 함수 형태로 만들어서 호출하실 필요가 없어 보입니다!